app/smc/pytest/e2e_stress.py

```diff
@@ -251,3 +258,89 @@ def test_upgrade_from_18x(tmp_path: Path, board_name, unlaunched_dut, arc_chip_d
         (15 << 16),
         (21 << 16),
     )
+
+
+def test_temperature_sensors(arc_chip_dut, asic_id):
+    test_name = "Temperature sensor test"
+    arc_chip = pyluwen.detect_chips()[asic_id]
+    total_tries = min(MAX_TEST_ITERATIONS, 100)
+    fail_count = 0
+
+    for _ in range(total_tries):
+        for sensor_id in range(NUM_TS):
+            response = arc_chip.arc_msg(
+                ARC_MSG_TYPE_READ_TS, True, False, sensor_id, 0, 5000
+            )
+
+            temp = response[2]
+            if temp < 40 or temp > 70:
+                fail_count += 1
+
+    report_results(test_name, fail_count, total_tries)
+    assert fail_count, f"{test_name} failed {fail_count}/{total_tries} times"
```

Comment:
This is only checking that `fail_count` is not `None`.

What we shoudl do here is validate that the failure rate is less than a specific number. I.e. that `fail_count / total_tries < max_failure_rate`